source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -187,6 +193,109 @@ public IEnumerable<PimsCompReqLeasePayee> GetCompensationRequisitionLeasePayees(
             return _compensationRequisitionRepository.GetCompensationRequisitionLeasePayees(compReqId);
         }
 
+        public PimsCompensationRequisition GetCompensationRequisitionAtTime(long compReqId, DateTime time)
+        {
+            return _compensationRequisitionRepository.GetCompensationRequisitionAtTime(compReqId, time);
+        }
+
+        public IEnumerable<PimsPropertyAcquisitionFile> GetCompensationRequisitionAcqPropertiesAtTime(long compReqId, DateTime time)
+        {
+            return _compensationRequisitionRepository.GetCompensationRequisitionAcqPropertiesAtTime(compReqId, time);
+        }
+
+        public IEnumerable<PimsPropertyLease> GetCompensationRequisitionLeasePropertiesAtTime(long compReqId, DateTime time)
+        {
+            return _compensationRequisitionRepository.GetCompensationRequisitionLeasePropertiesAtTime(compReqId, time);
+        }
+
+        public IEnumerable<PimsCompReqAcqPayee> GetCompensationRequisitionAcquisitionPayeesAtTime(long compReqId, DateTime time)
+        {
+            var acqPayees = _compensationRequisitionRepository.GetCompensationRequisitionAcquisitionPayeesAtTime(compReqId, time);
+
+            foreach (var payee in acqPayees)
+            {
+                if (payee.InterestHolder != null)
+                {
+                    var interestHolder = payee.InterestHolder;
```

Comment:
nit: could but this before the if statement did avoid dereferencing twice (also looks neater).